### ISSUE TITLE
Allow various parliament layouts

### ISSUE THREAD

**Issue by slashme** [slashme/parliamentdiagram#42]
This is a common request, and it would not be that difficult to code a tool that allows templates of parliaments to be made and filled in:

* The user gives a name (e.g. "Tinytown Parliament"), a list of coordinates of spots in the order that they should be filled, for example [(75.0,1.1),(1.1,1.1),(75.0,98.9),(148.9,1.1)] and a spot size (e.g. "1.0"). This would need to be designed by the user, because the script couldn't reasonably predict any possible layout. This might be saved on Wikipedia somewhere or on a user page, or might be uploaded by the user as JSON.  
* The user also gives a list of parties and their support, e.g. [("Speaker","DarkGray",1),("Party party","Red",1),("No-party party","#EED211",2)] - in this case, the "speaker" spot would be in the middle on the bottom, the "Party party" would be at the left on the bottom, and the other two spots would be the "No-party party"  
* The script then reads the layout file, checking that it's valid (e.g. no spots outside the diagram border, and no more than some reasonable number of spots). It then fills in the spots in the order given, and writes an SVG.

I'm not sure that this is a worthwhile goal, but if you feel strongly about it, add your voice to this feature request on GitHub and I could certainly code it.

**Comment by Gouvernathor** [comment]
Note that this would only work with layouts with a fixed number of seats. The way the arch or the westminster add or remove rows, and the way the arch changes the number of seats in rows, cannot be implemented using that kind of template.

**Comment by slashme** [comment]
Correct. This would only be for parliaments with a fixed layout, so that a seating plan can be simulated.
